spec/spec_helper.rb

```diff
@@ -93,15 +94,31 @@
   # Geocoding
   config.before(:each) { Spree::Address.any_instance.stub(:geocode).and_return([1,1]) }
 
+  default_country_id = Spree::Config[:default_country_id]
+  checkout_zone  = Spree::Config[:checkout_zone]
+  currency = Spree::Config[:currency]
   # Ensure we start with consistent config settings
-  config.before(:each) { Spree::Config.products_require_tax_category = false }
+  config.before(:each) do
+    reset_spree_preferences do |spree_config|
+      # These are all settings that differ from Spree's defaults
+      spree_config.default_country_id = default_country_id
+      spree_config.checkout_zone = checkout_zone
+      spree_config.currency = currency
+      spree_config.shipping_instructions = true
+      spree_config.auto_capture = true
+      spree_config.allow_backorders = false
+    end
```

Comment:
does it really need to happen for every single test example? this is going to add a considerable overhead to all the test suite.

Reply:
I agree that it is not ideal, but it does not seem to affect total spec runtime significantly. Spree's approach clears the preference cache before each spec, so there is no easy way to use customised default values without a) resetting them for each spec, or b) overriding the defaults set in Spree's `core/app/models/app_configuration.rb`.

I will try think of another way of achieving the same outcome.

Reply:
> so there is no easy way to use customised default values without a) resetting them for each spec, or b) overriding the defaults set in Spree's core/app/models/app_configuration.rb.

Then there's not much else we can do here. We need to solve this problem anyway so I'd move forward even with any performance impact.

I just raise the issue because this is what I'd check for if I were to speed up the test suite. Normally, you can apply the particular settings only to the examples that require it by means of https://relishapp.com/rspec/rspec-core/docs/metadata/user-defined-metadata.

Reply:
Yeah, we could try and work out which examples actually need which these 'defaults', but then we have the problem of polluting our test suite with settings that we should be able to assume are in a particular state. They are all set during app initialisation so allowing to default to something else for tests is confusing. I'd prefer to work out a way of setting these six preferences in a universal way, and only adding deviations from the assumed 'default' settings for a given example.